tools/install-monorepo.sh

```diff
@@ -15,143 +17,344 @@ function usage {
+			sudo apt install git || die "Unable to install git!"
+			echo "Installed git."
+		else
+			die "Installer script requires git to be installed."
+		fi
 	fi
-else
-	info "Updating brew"
-	brew update
-	# Brew can be finicky on MacOS
-	if [[ $? -ne 0 ]]; then
-		 echo "Reinstalling Homebrew"
-		 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+	echo "git: available"
+
+	if ! has_command curl; then
+		echo "curl: not found"
+		if [[ "$OS" == "Linux" ]]; then
+			echo "Installing curl..."
+			sudo apt install curl || die "Unable to install curl!"
+			echo "Installed curl."
+		else
+			die "Installer script requires curl to be installed."
+		fi
 	fi
-fi
+	echo "curl: available"
+}
 
-info "Checking if NVM is installed..."
-if ! command -v nvm &>/dev/null; then
-	info "Installing nvm"
-	curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash && export NVM_DIR=$HOME/.nvm && source $NVM_DIR/nvm.sh  --no-use
-else
-	info "Updating nvm"
-	curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
-fi
+# Checks for, installs, and updates Homebrew.
+function do_homebrew {
+	if ! has_command brew; then
+		echo "brew: not found"
+		echo "Installing brew..."
+		/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" || die "Unable to install brew!"
+		if [[ "$OS" == "Linux" ]]; then
+			HOMEBREW_PREFIX="/home/linuxbrew/.linuxbrew"
+		else
+			HOMEBREW_PREFIX="/opt/homebrew"
+		fi
 
-# Install and use the correct version of Node.js
-info "Installing Node.js"
-nvm install && nvm use
+		# Add brew env to current script for use later.
+		eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"
 
-# Install our requirements
-info "Checking Bash version..."
-if [[ -z "${BASH_VERSINFO}" || -z "${BASH_VERSINFO[0]}" || ${BASH_VERSINFO[0]} -lt 4 ]]; then
-	brew install bash
-fi
+		if ! has_command brew; then
+			die "Unable to install brew!"
+		fi
+		echo "Installed brew."
 
-info "Checking if jq is installed..."
-if ! command -v jq &>/dev/null; then
-	info "Installing jq"
-	brew install jq
-fi
+		# Determine shell RC file.
+		# Adapted from Homebrew install script:
+		# https://github.com/Homebrew/install/blob/master/install.sh
+		case "${SHELL}" in
+			*/bash*)
+				if [[ "$OS" == "Linux" ]]; then
+					shell_rcfile="${HOME}/.bashrc"
+				else
+					shell_rcfile="${HOME}/.bash_profile"
+				fi
+				;;
+			*/zsh*)
+				if [[ "$OS" == "Linux" ]]; then
+					shell_rcfile="${ZDOTDIR:-"${HOME}"}/.zshrc"
+				else
+					shell_rcfile="${ZDOTDIR:-"${HOME}"}/.zprofile"
+				fi
+				;;
+			*/fish*)
+				shell_rcfile="${HOME}/.config/fish/config.fish"
+				;;
+			*)
+				shell_rcfile="${ENV:-"${HOME}/.profile"}"
+				;;
+		esac
+
+		# Add brew init to shell rc file.
+		if ! grep -qs "eval \"\$(${HOMEBREW_PREFIX}/bin/brew shellenv)\"" "${shell_rcfile}"; then
+			(echo; echo 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"') >> ${shell_rcfile}
```

Comment:
This also doesn't work right.
```suggestion
			(echo; echo "eval \"\$(${HOMEBREW_PREFIX}/bin/brew shellenv)\"") >> "${shell_rcfile}"
```